### PR TITLE
DEPS: bump pyarrow version to 0.17.0 #38870

### LINKED ISSUE #38870: DEPS: bump pyarrow min to 0.17.0
this would remove some complexity around parquet, csv engine & string array. this is an optional dep and so i don't think this a big deal.

keep in mind that we likley won't release 1.3 until june time frame anyhow

### PR DESCRIPTION
**Pull request by fangchenli**
closes #38870

### CODE REVIEW

**Comment by jreback** [comment]
lgtm ping when green.

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -806,22 +804,10 @@ def test_additional_extension_arrays(self, pa):
...
         )
-        if Version(pyarrow.__version__) >= Version("0.16.0"):
-            expected = df
-        else:
-            # de-serialized as plain int / object
-            expected = df.assign(
-                a=df.a.astype("int64"), b=df.b.astype("int64"), c=df.c.astype("object")
-            )
-        check_round_trip(df, pa, expected=expected)
+        check_round_trip(df, pa, expected=df)
```
Comment (jorisvandenbossche) [inline pandas/tests/io/test_parquet.py:807]:
```suggestion
        check_round_trip(df, pa)
```

If it's `df`, it doesn't need to be specified as `expected` (and the same for the case below)

File: ci/deps/actions-37-db-min.yaml [hunk 2]
```diff
@@ -31,7 +31,8 @@ dependencies:
   - openpyxl
   - pandas-gbq
   - google-cloud-bigquery>=1.27.2 # GH 36436
-  - pyarrow=0.17 # GH 38803
+  - protobuf>=3.12.4
+  - pyarrow=0.17.1 # GH 38803
```
Comment (fangchenli) [inline ci/deps/actions-37-db-min.yaml:35]: Using `pyarrow=0.17.0` would pin `protobuf` version and cause an AttributeError in gbq test.

File: ci/deps/actions-37.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ dependencies:
   - numpy=1.19
   - python-dateutil
   - nomkl
-  - pyarrow=0.15.1
+  - pyarrow
```
Comment (fangchenli) [inline ci/deps/actions-37.yaml:21]: Using `pyarrow=0.17.0` causes all tests using `lxml` to fail, not sure why.

**Comment by jreback** [comment]
thanks @fangchenli very nice!

**Pull request merged**